[WIP] Dev zdcrec. New format for RecEvent

cortesep opened the pull request:
First version with updated format for reconstructed events

mwinn2 committed:
added ROF to MCH

fix eventtime

fix rof

include MCH dataformats to allow for ROF

compiling version

digitisation in rofs

fixes

add rofs to output

typo

added rofs in header of Digit Writer

added rofs in header of Digit Writer 2

naming convention MCLabels

added link for rof

fix rofloops, start digit-reader for simulation workflow

include reading from MCH digits for workflow

linking, header placing

linking

arguments

added Digit reader

remove couts

fill ROFRecord properly

fixed format

change naming, fix loop for rof in Stee

rebasing finalised

simplification

sawenzel committed:
Introducing simulation as a service

This commit brings the possibility to run
"o2-sim" as service/daemon, in the sense that its
lifetime extends that of a simple production of events.

This has the advantage that we may initialize the simulator
once for the simulation of many (different) batches of events.

The main benefits are:

a) Reduced initialization overhead for
   - transport of multiple timeframes
   - transport of multiple signal types
     (which can be combined in digitization)

   See example /run/SimExamples/SimAsService_basic

b) Enabling efficient biasing with the stop-go mechanism.
   Use the service in multiple stages:
   - transport part of event
   - trigger on output
   - continue with rest of triggered events

   See example /run/SimExamples/SimAsService_biasing1

The new mechanism is used via:

1. Launch and init o2-sim in service mode:

   - `o2-sim --asservice -e TGeant4 ...`

2. A separate client (python) script that talks to
   the service (via ZMQ):

   - `o2-sim-client.py "-n 20 -g pythia8 -o foo`

Some accompanying changes here are

* restructuring of o2sim json setup
* allowing to define hooks acting on generator trigger decisions
* some restructuring of communication between actors (needs further improvement)
* sim workers now have a notion of their ID

Note that the feature can be used. But further improvements (stability, error handling)
are needed and foreseen.

noferini committed:
TOF channel calibration using cosmics

add on-the-fly tracking for cosmics crossing many strips in a TOF module

add TOF calibration with cosmics

update for calibration with cosmics

templetization of TOF channel calibration to read different objects

Improvements and fixes

format

Possibility to calibrate not only at the EoR.

We need still to change the logic, because now it happens that
too few pairs have enough entries for the fit, so in the end we
close a time slot, but without being able to produce a CCDB
entry.

mwinn2 committed:
CLANG format

mwinn2 committed:
digits reader from dev

mwinn2 committed:
remove comments from MID

mwinn2 committed:
remove outcommented timing related information

ktf committed:
DPL: improve signaling logic

The fact that lastSignal was global introduced a race condition
for the second (third...) reader. This now makes it per device so
that we do not skip signals for different devices which come too close.

Also add metrics to debug the time between different signals.

chiarazampolli committed:
Possibility to calibrate with 1 slot valid till infinity and update when ready

TimeSlewing uses now 1 slot only valid till infinity

A few fixes + new feature

New feature allows to define after how many TFs to calibrate in
case one sets that the time slot length is infinite. In this
way, it won't try to calibrate at every TF, but with a resonable
frequency.

Improving calibration update when slot is infinite-wide.

clang-format

README updated

Typo fixed

mwinn2 committed:
fixed code checker complaints

aphecetche committed:
Update MCHDigitWriterSpec.h

knopers8 committed:
[OCTRL-355] AliECS dump: Use global channel space for raw channels

chiarazampolli committed:
Possibility to retry to calibrate with a lower frequency than the update interval

Case of mCheckIntervalInfiniteSlot = 0

clang-format

Fixes from review

Fix

clang-format

braces

space

braces again

clang-format

shahor02 committed:
Check/enforce that all links got AddData call in every IR it processed

RawFileWriter will check for every IR for which at least 1 link was filled by detector
if all other registered links also got AddData call. If not, it will be enforced internally
with 0 payload and trigger and detField values provided in the 1st addData call of this IR.
This may be pose a problem in case detector needs to insert some header even for 0-size payloads.
In this case it is detector responsibility to make sure that all links receive their proper
addData call for every IR.

knopers8 committed:
AliECS: reduce the resource requirements to the minimum

For now we don't have means to estimate the processing needs of the processes,
so we assume the lowest possible to avoid being blocked when running large
workflows on VMs.

knopers8 committed:
AliECS dump: support RPMs generated the new way

knopers8 committed:
AliECS dump: parametrize --resources-monitoring

davidrohr committed:
GPU: Add a cache for CUDA runtime compilation results

davidrohr committed:
GPU: Fix headers for CUDA RTC compilation

shahor02 committed:
make sure MFT digi2raw creates 1 file per link if requested

mfasDa committed:
[EMCAL-699] Add RCU trailer also for links without payload

In case a link does not contain data for a given
event, still a RCU trailer with the SRU configuration,
L1 phase and payload size 0 needs to be added
according to the implementation in the SRU
firmware.

davidrohr committed:
Full system test: Adapt to some renaming in DD

shahor02 committed:
path for collisioncontext can be set via NameConf.mDirCollContext

shahor02 committed:
Do not start CTFwriter if detectors mask is empty

njacazio committed:
PWGHF: update QA task for efficiency (#6023)

- Add event selection to efficiency computation (discarding events with no contributors)
- Simplified efficiency task
- Add pT cut compliant with histo titles
- Update task names
- Selecting Z position of gen vertices

Commit:
New MFT Power cables and services update

davidrohr committed:
GPU: AMD HIP creates a cache variable, overriding our settings if not FORCEd

davidrohr committed:
GPU: Check that define for HIP warp size is correct

AllaMaevskaya committed:
channel offsets in reconstruction (#6115)

* channel offsets in reconstruction

* move set-up CCDB to workflow

shahor02 committed:
Add MaxDXYIni cut on DCAFitterN seed XY DCA

robincaron13 committed:
Add MFT alignable volume functions (#5974)

* Add alignable volume functions

* Fix clang

* Fix conversion ID when extracting NumberOfSensorsPerLadder

* Fix chipID for alignable entry

* Restoring files that should not be touched by this PR

* Restoring files that should not be touched by this PR (bis)

* Small fix

Co-authored-by: Rafael Pezzi <rafael.pezzi@cern.ch>

shahor02 committed:
throw exception with message instead of cout + exit

julekm committed:
implementation of active watching folder for files with tracks

afurs committed:
FIT: Large FITRaw upgrade (#6152)

* FIT: isTCM method added to LUT

* FIT: RawReader class refactored

* FDD: executable for raw2digit tests added

* FIT: LUT and FV0 digit preparation

* FIT: FITRaw library refactoring

FIT: FITRaw library refactoring

FIT: FITRaw library refactoring

* FIT: FV0 preparation for FITRaw lib, and some fixes in LUT

* FV0: FITRaw lib applied on FV0

FIT data formats: DPL channel name added

FIT: workflow lib added and applied to FV0

FV0: FITRaw lib applied on FV0

* Update FITDigitWriterSpec.h

* Minor changes

* some fixes

shahor02 committed:
Add cosmics to RecoContainer and InputHelper

shahor02 committed:
PVertexing: optional afterburner for track-vertex reattachment

+ multiple fixes for debris reduction
+ extend optional debug output for parameters tuning.
+ somewhat technical note in the README on what/how should be tuned.

ktf committed:
DPL: implement ability for devices to watch driver messages

davidrohr committed:
Compatibility to boost 1.76

davidrohr committed:
GPU: Add define GPUCA_GPUCODE_HOSTONLY: disables inclusion of GPU device-code headers and duplacte compilation of default kernels in host-only source files where they are not used

davidrohr committed:
Move SHA1 algorithm in 3rdparty header only library

davidrohr committed:
GPU: Use SHA1 hashes to check if RTC cache is outdated

davidrohr committed:
GPU: Prepend some more macros with GPUCA_

davidrohr committed:
GPU: RTC works in combination with non-constant memory

davidrohr committed:
GPU: Use dummy member shadowing constexpr-replaced members to maintain same memory layout

pillot committed:
fix workflow option (#6142)

dstocco committed:
[MID] Executable to compare the User Logic raw data with the common logic one (#6144)

davidrohr committed:
GPU: Split host and GPU code during CUDA compilation, mostly for faster compilation

mario-krueger committed:
DPL Analysis: store created histograms in unique_ptrs (#6159)

shahor02 committed:
Fix in VertexTrackMatcher ITS track time treatment

davidrohr committed:
GPU: Automatically disable device timers if not supported by OpenCL platform

davidrohr committed:
GPU: Bugfix for OpenCL1 processing / GPU processing with track merging on host

vkucera committed:
Add tab detection to CI tests. (#6099)

* Add tab detection to CI tests.

* Improve diffing.

shahor02 committed:
Add FT0 Channels to RecPointReader and RecContainer

davidrohr committed:
GPU: Generate GPU kernel definitions dynamically, and don't precompile them in the RTC source

davidrohr committed:
GPU: Move some more functions from GPUReconstructionCUDAKernels.cu to other files

sawenzel committed:
Use correct type in sim duplication tool

Were using wrong types in kinematics duplication.
This commit fixes this.

shahor02 committed:
Add default c-tors to PVertex debug dump objects

davidrohr committed:
GPU: The CMake OpenMP module apparently doesn't know about CUDA, so it is not adding the openmp flat

pbuehler committed:
Cleaning and updates of tutorials (#6158)

jgrosseo committed:
Fix comment

pbuehler committed:
updates of the O2 Analysis Framework shall from now on be documented at
https://github.com/AliceO2Group/analysis-framework

pzhristov committed:
Changes for Root v6-24 (#6166)

* Changes for Root v6-24
- Test some macros in compiled mode
- Corrected include files
- Define the ms_gsl version

Co-authored-by: hristov <Peter.Hristov@cern.ch>

davidrohr committed:
GPU: Add option to run parallel per-kernel RTC compilation

davidrohr committed:
GPU: If we have a specialization for a kernel, we can skip compiling the base one to save some compile time

chiarazampolli committed:
PWGHF: Implementation of Lc --> K0s+p (#5257)

* First version of skimming for Lc

some changes

update

Expect that v0s are already built and preselected

some more work

A bit further, but not compiling

just a temp commit

Compiling version - still refinements needed

further work

temp

not working

not working again, needed for debug

ppp

working version - cuts still to be tuned

clang

few changes

clang

further work

forgotten file

forgotten file

Fixing copy/paste error

clang

* Selector for Lc --> K0s+p added

clang-format

* First version of histogramming task - not compilinggit add Analysis/Tasks/PWGHF/CMakeLists.txt Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx Analysis/Tasks/PWGHF/taskLcK0sp.cxx

fix compilation

fixes

clang

* Avoid passing indices of daughters when building v0

In case, one would need a o2::dataformats::GlobalTrackID.h, but anyway it is not
used.

Fix

* Comments by Vit

clang

Fixes for braces

clang

* Adding pT V0 daugh, and swapping V0 and bach (as in AliPhysics)

clang

* First version using MC info.

To be investigated:
- why no V0 in the V0 table is a K0s that comes from a Lc?
- the RecoDecay::isMatchedMCGen might need some adjustments, being checked

Fixes

Format

* recent work - to be cleaned-up

* Update Lc -> K0S+p to do MC checks too.

Adding also the possibility to do local debugging with an AO2D file from
any production.

format

Stupid fix

* Revisiting debugging

* Anton's comments

clang-format

* Implementation of comments by Anton and Vit

misc fixes

clang-format

* more comments by Vit

clang-format

* Further comments implemented

clang-format

leftovers fixed

Clang-format

Few more

Clang-format

removing useless namespace declaration

Optimization

more

clang-format

typo

Remove commented out code

julekm committed:
saving tracks from workflow to jsons

jgrosseo committed:
Fix filter (#6171)

rpezzi committed:
[MFT] Standalone vertex matching (time-bracket) & AOD table production (#6120)

* Including MFT tracks on RecoContainer

Conflict of fwd tracks at GPUWorkflowHelper.cxx

* Isolate Central Barrel and Forward tracks on createTracksVariadic

* MFTTracks-Vertex time-bracket matching

* Restore RecoContainer::createTracks

* Rebase MFT Track-Vertex matcher

* Use vector as MFT tracks MCLabels instead of MCTruthContainer

For full compatibility with RecoContainer

* Add MFT Standalone tracks on AOD producer workflow

* Fix ROF time and error

Commit:
moved Calib objects to DataFormats; Reimplement clusters as obj. with cluster properties+ref to separate list of CluElements used to (re-)calculate E, position etc. Re-implemented cluster unfolding procedure and turn it on by default

shahor02 committed:
Log statistics on track to vertex assignment

shahor02 committed:
Fix in composition of ITS/MFT dictionary name

davidrohr committed:
Add missing ROOT dictionary (not really needed, but fixes warning)

matthiasrichter committed:
Have to use FATAL message to catch failure condition in workflow unit tests

At some point the logic in the DPL driver has changed and the workflow
does not terminate with an error code if a device has messages of severity
ERROR.

shahor02 committed:
Fix in ITS digits->raw conversion

Due to the problem in the cables mapping the digits of half-stave 0 on the ITS
layers 3 and 4 were not encoded/decoded

mario-krueger committed:
Analysis: remove deprecated HistHelpers (#6176)

davidrohr committed:
Fix HMPID stream decoder workflow writing output at end of run not at endOfStream

shahor02 committed:
FT0 correct treatment of missing TF

shahor02 committed:
correct treatment of missing TF in HMPID DataDecoderSpec

davidrohr committed:
GPU configKeyValues: support nested settings with '_' syntax as separator

davidrohr committed:
Nested the lightweight and heavyright runtime settings of the GPU display in GPUSettingsDisplay

vkucera committed:
Add detection of trailing whitespaces.

strogolo committed:
Software event filtering for Run 3 pp (#5967)

rpezzi committed:
[MFT] Marking reconstructed tracks on the MC AOD table

tklemenz committed:
minor fixes in TPC CalArray and TPC/qc PID

mpuccio committed:
Cluster finder for ITS3 (#6129)

* Adding clusteriser workflow for ITS3 and related changes

Co-authored-by: Maximiliano Puccio <mpuccio@users.noreply.github.com>

* First version of clusterfinder workflow ITS3 -- to be debugged

Co-authored-by: Maximiliano Puccio <mpuccio@users.noreply.github.com>

* First working version of ITS3 cluster finder

Co-authored-by: Maximiliano Puccio <mpuccio@users.noreply.github.com>

* Fixes for CheckClustersITS3 macro

Co-authored-by: Maximiliano Puccio <mpuccio@users.noreply.github.com>

* Introduce ITS3 compressed cluster format

* Extend mask size in PixelData for upgrades

* Use the same mask for ITS3 and ITS2

* Fix warning and hopefully macro linking problem

* Remove tabs?

* Add copyright banner to TopologyDictionary.cxx

* Add topology dictionary to the dictionary

* Remove macro compilation for osx

* Fix copy paste error in the CMakeLists

* Add TopologyDictionry to LinkDef

Co-authored-by: Alexander Kalweit <alexander.philipp.kalweit@cern.ch>

ktf committed:
Dropping old / obsolete / unmaintained code

Taking ownership in WP4 for the test_O2Device.cxx test
and renaming in to test_FairMQ.

mguilbau committed:
Adjusting the muon iterator tutorial analysis task (#6143)

* Adjusting the tutorial task to iterate on muons. It is now using FwdTracks

* Update the AODReaderHelpers to enable Fwd track table. Mandatory for the muon tutorial task to run

* Please consider the following formatting changes

* fixing the PR by uncomenting piece of the code

Co-authored-by: ALICE Action Bot <alibuild@cern.ch>

dsekihat committed:
Analysis/Task/PWGDQ: update v0selector (#6160)

* Analysis/Task/PWGDQ: update v0selector

* Analysis/Tasks/PWGDQ: remove unnecessary lines

* remove unnecessary variables

remove unnecessary variables

* fix for braces

fix for braces

* remove whitespace

remove whitespace

aferrero2707 committed:
[MCH] added HB packet validity check in UL decoder (#6192)

aferrero2707 committed:
[MCH] improved time frame grouping in CRU page reader (#6193)

shahor02 committed:
Fix for a typo in FT0 RecPointReaderSpec (from N.Burmasov)

chiarazampolli committed:
CCDB populator will match payload and wrapper according to DataDescription (#6150)

* changes for wildcarded DataSpec

* Using constructor of InputSpec with only Origin

* Modify detector calib.code to match to new CCDB populator inputs (#32)

Co-authored-by: shahoian <ruben.shahoyan@cern.ch>

* Adapting new devices

clang-format

* Removing obsolete DCS processing

* Adding some pragma rules

* remove deleted macro from exclusion list

* Remove tabs

* Include unordered_map to test if it helps with dictionary building

* Move include and pragma outside the ifdef (Giulio's advice)

* Update DetectorsDCSLinkDef.h

* Moving include of unordered_map to an include file in the library

* clang-format

* remove trailing space

Co-authored-by: Ruben Shahoyan <shahor02@users.noreply.github.com>
Co-authored-by: shahoian <ruben.shahoyan@cern.ch>
Co-authored-by: Giulio Eulisse <10544+ktf@users.noreply.github.com>

pillot committed:
[MCH] prepare setup for run3 (#6207)

sawenzel committed:
o2-sim: Improve communication between primary server and transport workers

More robust communication between primary servery and sim workers:

a) answer req-rep primary channel immediately (even if no work)
   - avoid blocking in request handling
   - also don't need to specify timeouts anymore
b) include server status information as header
   - this may be used by workers to back-off/retry

This solves some stability issues with the sim-as-a-service.
The 2 examples SimAsService_basic / SimAsService_biasing1 now run
stably.

ktf committed:
DPL: use Resource Manager to optimize shared memory usage (#6211)

AOD reader now will require some shared memory quota to be given
to it before it starts doing anything.

The system will keep track of how much shared memory was allocated,
and report back when it's being released (in form of a metric).

For the moment the maximum amount of shared memory available to the readers is 2GB, but it will be configurable in a subsequent PR.

ktf committed:
DPL: hotfix for resource manager (#6215)

* Demote some messages to debug.
* Properly handle more than one reader.

shahor02 committed:
sim_challenge: fix in VZ generation and ITS reco options in pp

preghenella committed:
TOF: fix retrieval of integer output buffer CompressorTask.cxx (#6194)

To fix retrieval of integer output buffer size output buffer size will be the sum of the command-line-specified buffer size and the value automatically calculated.
Unless a negative value is used as the command-line-specified buffer size, in which case a fixed buffer size is used.

vkucera committed:
Improve space checker report. (#6219)

- Removed neighbouring lines to make the log more compact.
- Added column numbers in the tab reports.
- Added tips on how to fix the errors and how to configure the editor to avoid them.
- Added workflow commands for highlighting.

cortesep committed:
Reworking event structure

aphecetche committed:
[O2-2018] Drop support for Microsoft GSL v2 (#6075)

* [O2-2018] Drop support for Microsoft GSL v2

Only use V3 from now on.
We can thus drop our custom `Findms_gsl.cmake`.
Also using the "official" target name of `Microsoft.GSL::GSL` instead of
our own `ms_gsl::ms_gsl`.

* chasing recent mods

pbuehler committed:
Further updates of tutorials (#6191)

* . updated names of tutorial tasks
. removed TaskName from adaptAnalysisTask
. some minor cleanings of tutorials

* In this case the templated task requires a TaskName

* to resolve conflict

Co-authored-by: jgrosseo <jgrosseo@cern.ch>

ktf committed:
DPL: properly handle SIGTERM (#6218)

ktf committed:
DPL: add test for ability to create transport (#6212)

cortesep committed:
Corrected compilation bug

sawenzel committed:
Create README.md

reference analysis documentation available in the top-level analysis folder

ekryshen committed:
Event selection adapted to Run2 and Run3 MC (#6220)

cortesep committed:
New ZDC rec event format

alibuild committed:
Please consider the following formatting changes

wiechula committed:
missing includes to compile with gcc11

davidrohr committed:
GPU: Option renaming, and grouping of TRD and TPC options

davidrohr committed:
GPU: Add separate config structure for RTC, and check whether RTC options have changed for RTC cache

ktf committed:
Do not use printf (#6232)

shahor02 committed:
Suppress leftover PHOSCalib directory [fix for PR6187]

shahor02 committed:
fix in CCDB populator documentation

sawenzel committed:
simlogcheck: cat problematic files

print problematic file so that we can see the problem in the CI logs

Commit:
BadChannelsMap renamed

mario6829 committed:
Remove fake IB HalfStave and Module volumes (#6011)

* Remove fake IB HalfStave and Module volumes

* Change IB service container name (to avoid "Module")

* Adding dummy commit to trigger anothe CI check

* RecoGeomHelper: Fix calculation of nLadders on inner layers

* Fix getChipIndex for inner layers

Co-authored-by: Mario Sitta <Mario.Sitta@cern.ch>
Co-authored-by: Ruben Shahoyan <shahor02@users.noreply.github.com>
Co-authored-by: shahoian <ruben.shahoyan@cern.ch>

shahor02 committed:
rANS: allow encoders/decoders with empty dictionaries (#6202)

* rANS: allow encoders/decoders with empty dictionaries

saganatt committed:
Update histogram registry spec to make it more distinguishable (#6238)

aalkin committed:
DPL Analysis: Handle unassigned (-1) index in grouping (#6225)

ihrivnac committed:
Fixes in doxygen: (#6237)

* Fixes in doxygen:
Added doxygen keywords where missing

* Removed trailing whitespaces

hahassan7 committed:
[EMCAL-518,EMCAL-694] Split digitizer into SDigitizer and FEEDigitizer

shahor02 committed:
Fix: QED interaction sampler was using default BC scheme

shahor02 committed:
Fix parsing for InputSpec with DataOrigin only (#6241)

Co-authored-by: shahoian <ruben.shahoyan@cern.ch>

martenole committed:
[TRD] remove tabs and trailing whitespaces

noferini committed:
add tof file proxy for data w/o rdh

fgrosa committed:
PWGHF: implement event selection in track-index skim creator (#6016)

* Add workflow for event tagging

* Remove placeholder for event selection

* Add missing header

* Add histo to monitor selected/rejected events

* Fix format

* Fix fill of event histo and change struct name

* Add missing if

* Change struct names to avoit TaskName argument

* Use camelCase

* Use camelCase

* Update name of  structs

* Remove output histogram

* Update workflows with new struct names

* Revert removal of histogram

* Fix conflicts

* Fix format

davidrohr committed:
Grep logfile always as ascii file not as binary in jobutils.sh

davidrohr committed:
GPU: Add more missing ROOT dictionaries

ktf committed:
DPL Foundation: add C++20 version for counting

ktf committed:
DPL Foundation: add helper to apply functions to member of a struct

This is actually all we need. We do not need to create tuple out
of the members of the tasks.

ktf committed:
DPL Foundation: do not use a tuple to store task fields.

We can either apply directly the function, or extract base classes
for each of the fields with this approach.

ktf committed:
DPL: drop try catch

davidrohr committed:
GPU: Fix MS_GSL in CMake for GPU standaloneb benchmark after migration to v3

preghenella committed:
pythia8 generator flag will start empty pythia8 config, make sure you configure it

preghenella committed:
Example pythia8 configuration for p-Pb

pillot committed:
use configurable parameters for MCH tracking

ktf committed:
DPL: improve workflow validation (#6256)

Complain if special characters are used as DataProcessor names.

ihrivnac committed:
More fixes in doxygen: (#6262)

* More fixes in doxygen:
Added doxygen keywords where missing, match the subpage references

* Removed trailing whitespaces

vkucera committed:
PWGHF: Fix track selection in index skimming (#6255)

mcoquet642 committed:
MFT Noise calibration (#6230)

* First version of MFT calibration : generating noise map and writing in CCDB

* Update README.md

* modifié :         Detectors/ITSMFT/MFT/calibration/CMakeLists.txt
	nouveau fichier : Detectors/ITSMFT/MFT/calibration/include/MFTCalibration/NoiseCalibratorDigits.h
	nouveau fichier : Detectors/ITSMFT/MFT/calibration/include/MFTCalibration/NoiseCalibratorDigitsSpec.h
	modifié :         Detectors/ITSMFT/MFT/calibration/src/MFTCalibrationLinkDef.h
	nouveau fichier : Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorDigits.cxx
	nouveau fichier : Detectors/ITSMFT/MFT/calibration/src/NoiseCalibratorDigitsSpec.cxx
	modifié :         Detectors/ITSMFT/MFT/calibration/testWorkflow/README.md
	modifié :         Detectors/ITSMFT/MFT/calibration/testWorkflow/mft-calib-workflow.cxx

* modifié :         Detectors/ITSMFT/MFT/calibration/testWorkflow/mft-calib-workflow.cxx

* Implementation of noise calibration DPL using digits

* Update README.md

* TimeSlot calibration

* Clang format

* Fix typo

* Fix typo

* Fix typo

* Update NoiseCalibrator.h

* Update NoiseCalibratorDigits.h

* Updating TimeSolt implementation

* Clang format

* Digits/Clusters implementation

* Using StringUtils

* Clang Format

* Fixing spaces and tabs

* Changes for PR

	modifié :         ../include/MFTCalibration/NoiseCalibrator.h
	modifié :         ../include/MFTCalibration/NoiseCalibratorSpec.h
	modifié :         ../include/MFTCalibration/NoiseSlotCalibrator.h
	modifié :         NoiseCalibrator.cxx
	modifié :         NoiseCalibratorSpec.cxx
	modifié :         NoiseSlotCalibrator.cxx

* Clang format

* Changes for PR

* Clang Format

* Fix typo

Co-authored-by: Maurice <maurice.coquet@yahoo.com>

cortesep committed:
Small updates

cortesep committed:
clang format

alibuild committed:
Please consider the following formatting changes

pbuehler committed:
Added python tool to create html representation of O2 data model (#6243)

ktf committed:
DPL Analysis: send the size of the tables as metric (#6217)

shahor02 commented on DataFormats/Detectors/ZDC/include/DataFormatsZDC/RecEvent.h:
```diff
@@ -59,7 +86,6 @@ template <>
 struct is_messageable<o2::zdc::RecEvent> : std::true_type {
```
RecEvent is not messegeable, since it contains dynamic structures as data members.

cortesep replied:
Removed

shahor02 commented on DataFormats/Detectors/ZDC/include/DataFormatsZDC/RecEvent.h:
```diff
@@ -12,36 +12,63 @@
+  }
   void print() const;
-
   ClassDefNV(RecEvent, 1);
```
Since it does not need to be persistent but is just a wrapper, you don't need dictionary (and adding it to LinkDef)

cortesep replied:
Leaving it for the moment. Introduced a note to remove it when going into production

shahor02 commented on DataFormats/Detectors/ZDC/include/DataFormatsZDC/RecEvent.h:
```diff
@@ -12,36 +12,63 @@
 namespace zdc
 {
-
 struct RecEvent {
```
In general, the concept of `mRecBC` referring to ranges in the `mEnergy`  etc. is orthogonal to having a RecEvent as a single object. Your `RecEvent` is in fact the result of the reconstruction for the full TF, and you should not store / send it as a single object. If you want, you can keep it as a wrapper for its vectors in the reconstruction code, to make it more readable, but the reco. spec. should send these individual vectors to profit from shared memory transport.

shahor02 commented on DataFormats/Detectors/ZDC/include/DataFormatsZDC/RecEventAux.h:
```diff
@@ -0,0 +1,106 @@
+///
+/// It does not fulfill is_messageable because the underlying ROOT
+/// classes of Point2D are note trivially copyable.
+namespace framework
+{
+template <typename T>
+struct is_messageable;
+template <>
+struct is_messageable<o2::zdc::RecEventAux> : std::true_type {
+};
+} // namespace framework
```
this is also not messageable. And, as far as I understand, this is just a debug output to be stored in the root tree, right?

cortesep replied:
Removed

shahor02 commented on Detectors/ZDC/reconstruction/include/ZDCReconstruction/RecoParamZDC.h:
```diff
@@ -0,0 +1,52 @@
+  Int_t beg_ped_int[NChannels] = {DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, Dummy ...
+  Int_t end_ped_int[NChannels] = {DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, DummyIntRange, Dummy ...
+  std::array<bool, NTDCChannels> bitset = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0}; // Set bits in coincidence
+ public:
```
the `struct` is public by default

shahor02 commented on Detectors/ZDC/reconstruction/src/DigiReco.cxx:
```diff
@@ -0,0 +1,785 @@
+
+  if (mTreeDbg) {
+    // Open debug file
+    mDbg = new TFile("ZDCReco.root", "recreate");
+    mTDbg = new TTree("zdcr", "ZDCReco");
```
You better use `unique_ptr`, see e.g.
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/include/DetectorsVertexing/PVertexer.h#L155-L156
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/src/PVertexer.cxx#L615-L616
https://github.com/AliceO2Group/AliceO2/blob/7a6c7f384bddc76e787b6efa32c5a0842c661cf1/Detectors/Vertexing/src/PVertexer.cxx#L638-L644

shahor02 commented on Detectors/ZDC/reconstruction/src/DigiReco.cxx:
```diff
@@ -0,0 +1,785 @@
+  }
+}
+
+int DigiReco::process(const std::vector<o2::zdc::OrbitData>* orbitdata, const std::vector<o2::zdc::BCData>* bcdata, const std::vector<o2::zdc::ChannelData>* chdata)
```
I assume this method is called from the DigitRecoSpec, passing the spans extracted from the inputs?
Then here also one should use e.g. `gsl::span<const o2::zdc::OrbitData> orbitData` etc.

cortesep replied:
Changed method to use span instead of vectors

shahor02 commented on Detectors/ZDC/workflow/src/DigitRecoSpec.cxx:
```diff
@@ -0,0 +1,92 @@
+  //   auto eeb = CTF::get(buffer.data()); // cast to container pointer
+  //   eeb->compactify();                  // eliminate unnecessary padding
+  //   buffer.resize(eeb->size());         // shrink buffer to strictly necessary size
+  //  eeb->print();
```
Here, after doing the reconstruction, you should extract the vectors which are currently the part of the RecEvent and snapshot them one by one, e.g.
````
pc.outputs().snapshot(Output{"ZDC", "BCREC", 0, Lifetime::Timeframe}, recEvent.mRecBC);
pc.outputs().snapshot(Output{"ZDC", "ENERGY", 0, Lifetime::Timeframe}, recEvent.mEnergy);
pc.outputs().snapshot(Output{"ZDC", "TDCDATA", 0, Lifetime::Timeframe}, recEvent.mTDCData);
````


And these outputs

shahor02 commented on Detectors/ZDC/workflow/src/DigitRecoSpec.cxx:
```diff
@@ -0,0 +1,92 @@
+  inputs.emplace_back("trig", "ZDC", "DIGITSBC", 0, Lifetime::Timeframe);
+  inputs.emplace_back("chan", "ZDC", "DIGITSCH", 0, Lifetime::Timeframe);
+  inputs.emplace_back("peds", "ZDC", "DIGITSPD", 0, Lifetime::Timeframe);
+
```
The outputs I mentioned above should be declared here, e.g.
````
std::vector<OutputSpec> outputs;
outputs.emplace_back("ZDC", "BCREC", 0, Lifetime::Timeframe);
outputs.emplace_back("ZDC", "ENERGY", 0, Lifetime::Timeframe);
outputs.emplace_back("ZDC", "TDCDATA", 0, Lifetime::Timeframe);
````

shahor02 commented on Detectors/ZDC/workflow/src/DigitRecoSpec.cxx:
```diff
@@ -0,0 +1,92 @@
+  return DataProcessorSpec{
+    "zdc-digi-reco",
+    inputs,
+    Outputs{{"ZDC", "RECODATA", 0, Lifetime::Timeframe}},
```
and used here

shahor02 commented on Detectors/ZDC/workflow/src/ZDCRecoWriterDPLSpec.cxx:
```diff
@@ -0,0 +1,42 @@
+  return MakeRootTreeWriterSpec(writerName.data(),
+                                fnameDef.data(),
+                                "o2rec",
+                                BranchDefinition<std::vector<o2::zdc::BCRecData>>{InputSpec{"RecBCinput", "ZDC", "RECBC"}, "ZDCRecBC"})();
```
here you should mention all inputs the writer should store, not only the RECBC

ktf committed:
Headers: remove unneeded traits (#6247)

Modern compilers will optimize.

ktf committed:
DPL: improve shared memory handling in AOD (#6240)

* Keep into account the provided input size
* Offer shared memory in incremental chunks, rather than all at once
* Fix logic to offer to the next reader

aferrero2707 committed:
MCH: introduce RoFRecord in decoder (#6109)

The MCH RoF are designed to group all SAMPA digits that
have exactly the same time stamp.
Further grouping of the digits into time clusters is
delegated to a later stage of the processing pipeline.

The code also introduces a unit test for the RoF
finding algorithm.

shahor02 committed:
DetID: add nameToID method

shahor02 committed:
dcs-config-proxy and sample consumer workflow

shahor02 committed:
Fix treatment of multiple raw FMQ channels

ktf committed:
DPL: rationalise parent-child communication (#6268)

vkucera committed:
Switch MC on by default (#6271)

njacazio committed:
Add flag to check if a track has TOF (#6266)

jgrosseo committed:
support unassigned indices (#6260)

shahor02 committed:
RawFileWriter can handle empty HBF fillers of >8KB

sawenzel committed:
Avoid inclusion of Vc/Vc in RandomRing

Inclusiong of Vc/Vc is argued to cause
gcc10 to take lot's of memory leading to crashes
in the CI.

In this particular case, I think we can avoid
the inclusion og Vc/Vc in this header and leave
it it to the source files to do so.
(But this might not always be possible);

pillot committed:
fix trailing whitespaces

sawenzel committed:
Treat FLUKA_VMC like a real plugin

We now don't need to know anything about Fluka_VMC
in order to compile with Fluka support in o2sim.
Setup is completely delegated to the runtime.

sawenzel committed:
o2sim: Cleanup tmp files

This should (finally) avoid spilling the tmp
folder with simulation socket leftovers.

ktf committed:
DPL DDS: use id rather than name for task id (#6276)

This will add support for the timePipelined devices.

davidrohr committed:
Full system test: disable shm-monitor for StfBuilder as is done automatically for the DPL processes

sawenzel committed:
Proper termination of detached thread

Wait for proper termination of detached thread,
and thereby prevent ZMQ socket problem.

shahor02 committed:
Do not use FairMCApplication::MisalignGeometry

Instead, we will start with ideal geometry and misalign it with GeomManager on the fly

ktf committed:
DPL: cleanup AODReaderHelpers.cxx (#6283)

RDataFrame is not actually needed.

pillot committed:
use configurable parameters for MCH clustering

shahor02 committed:
Fix: gpu::bitset prints bits from left to right (like std::bitser)

shahor02 committed:
Improve det.list to mask_t conversion

shahor02 committed:
Add standard alignment object path to NameConf

ktf committed:
DPL: reduce template instanciation (#6290)

ktf committed:
DPL: hide InputSpan from common headers

InputSpan usage of std::function causes an avalanche of template
instantiantions, resulting (at least) in a rather large slowdown
of Framework/Core compilation. With this patch we go from

 Compilation (452 times):
   Parsing (frontend):         1686.8 s
   Codegen & opts (backend):   1120.7 s

to

 Compilation (452 times):
   Parsing (frontend):         1441.4 s
   Codegen & opts (backend):    842.8 s

cortesep committed:
Some corrections from Ruben

shahor02 committed:
Fix in SVertices attachment to RecoContainer

ktf committed:
DPL: avoid dependency on boost for trivial stuff (#6289)

shahor02 committed:
Fix in requesting FT0 data for PVertex validation

jgrosseo committed:
documentation of data model. renaming unassigned tracks table (#6284)

njacazio committed:
PWGHF: Add single particle distribution (#5556)

- Extendend configurability of task
- Add possibility to set the PDG code of the particle of interest
- Remove '_' in variables
- Improve histo naming and axes
- Add even more configurability for the QA tasks
- Add PV contributors vs track mult plot
- Replace the qa Task with the simple one
- Fix camelCase
- Fix reference for resolution plots
- Add resolution of vertex in plots
- Fix spaces in CMakeFile

ktf committed:
Drop o2::byte (#6253)

Replace it with C++17 std::byte. This does not endorse the actual
idea behind std::byte usage, merely gets rid of an O2 specific type
and header for it.

mcoquet642 committed:
Fixing typo in VarManager (#6273)

Co-authored-by: Maurice <maurice.coquet@yahoo.com>

DelloStritto committed:
Add prompt-feeddown histo and fix LcpKpi selector (#5889)

sawenzel committed:
o2-sim: Improved signal handling

This should improve the shutdown phase
of the distributed simulation system upon errors:

*) now terminate using process groups in
   order to prevent leaking child processes
*) improved signal broadcasting in the Geant transport
   processes (within the process group)

ktf committed:
Add back cerrno (#6300)

aalkin committed:
DPL Analysis: fix AnalysisTask test after renaming data model table (#6301)

mario-krueger committed:
DPL Analysis: avoid instantiating HistFactory functions for each TU (#6302)

- move histogram creation code to HistogramSpec.cxx
- instantiate histogram creation templates only once

shahor02 committed:
Class to hold a bracket of InteractionRecords

shahor02 committed:
ITS tracking will produce output with reconstructed IRFrames (IR brackets)

The vector<IRFrame> will contain the 1st and the last IRs of the ITS strobe
which was reconstructed (in the sync. workflow conditioned by mult. selection)
and produced at least 1 track.

shahor02 committed:
Attach IRFrame to RecoContainer and InputHelper

Use:
dataRequest->requestIRFramesITS(); // in the device spec which needs it

o2::globaltracking::InputHelper::addInputSpecsIRFramesITS(configcontext, specs); // in the workflow building

alternatively, one can add the reader directly:

'#include "ITSWorkflow/IRFrameReaderSpec.h"'
...
specs.emplace_back(o2::its::getIRFrameReaderSpec());

recoContainer.getIRFramesITS() to access the span of IRFrame objects.

shahor02 committed:
Add unassigned tracks as last entry of Vtx to Track refs

The indices of unassigned tracks will be added in the end of the vector<VtxTrackIndex>
sent by the VertexTrackMatcher (PVTX_TRMTC DataDescription) and will be referenced
by an extra (last) element of the vector<VtxTrackRef> (sent as PVTX_TRMTCREFS DataDescription).

Therefore, for the TF with N vertices found the size of the vector<VtxTrackRef> will be N+1.

A getter VtxTrackRef::getVtxID() is added and it will return -1 for this special entry collecting
unassigned tracks

ktf committed:
DPL: stop at first workflow in the pipe which fails

This will also forward error messages from one workflow to the other, so that
we can properly visualise what actually failed.

shahor02 committed:
use track PID instead of fixed pion in tof correction

sawenzel committed:
CcdbApi: disable token checking on MacOS

Token checking has problems in MacOS (hangs in unit tests in some environments, etc).

Given that the mechanism is not relevant on this non-production platform,
let's simply disable it.

aalkin committed:
DPL Analysis: fix bug in unassigned grouping (#6275)

sawenzel committed:
o2sim: Harmonize socket names; fix rm command

shahor02 committed:
Modifications for alignment application to geometry

GeomManager::loadGeomerty(..) will trigger retrieval of alignment objects for all detectors from CCDB
and their application to all detectors (unless the applyMisalignment argument is set to false).
This process can be steered (CCDB to use, detectors list, timestamp) can be steered via
ConfigurableParam o2::base::Aligner, see O2/Detectors/Base/README.md

shahor02 committed:
Sample macros for dummy alignment objects uploading

macro/UploadDummyAlignment.C to populate CCDB with empty alignment containers for all detectors
Detectors/ITSMFT/ITS/macros/test/ITSMisaligner.C to upload ITS container with alignment objects
(with all parameters set to 0, so the geometry effectively will stay the same)

davidrohr committed:
GPU Display: improve configuration

shahor02 committed:
Generate header providing build-time ENABLE_UPGRADES status

If the O2 was built with ENABLE_UPGRADES, using installed headers dependin on the latter in the compiled macros
leads to the conflicts since the ACLIC is not aware of ENABLE_UPGRADES.
This PR will generate and install a header which defines ENABLE_UPGRADES if it was used at build-time.

cortesep committed:
Digits reader

sawenzel committed:
Some higher level filesystem utility

in particular a C++ way to get list files with globing

application to
* removal of tmp files in o2sim
* determining correct file input list for TPC chunked-digit merger

vfeuilla committed:
Updates for dimuon and Electron-Muon analysis (#6296)

* Update Worrkflow. Simplified TableReader without configurables for Hyperloop use

* Simplified TableMaker without configurables for Hyperloop use. Muon Tables in PbPb

* Adding Cuts for Muon analysis

* Modifying/Adding histograms for Dimuon analysis and Electron-Muon analysis

* Adding Electron-Muon case for the FillPair method

* Adding Electron-Muon Pair in the output histograms

* Simplified TableMaker without configurables for Hyperloop use, muon in PbPb

* Simplified TableMaker without configurables for Hyperloop use, muon in pp

* Update the workflow. Simplified TableReader without configurables for Hyperloop use

* Adding files for Hyperloop use

* removing trailing whitespaces

* Removing redundant file
Adding TableMaker_PbPb: duplicate of TableMaker for Hyperloop use in PbPb

* Small changes in the Muon Track Selection

* Correction of some errors in the Electron-Muon pairing

* Removing redundant file. Adding TableMaker_PbPb

* Clang formating

* Adding bits to the pair filterMap

* Correction on the electron selection in the Electron-Muon pairing

* Removing obsolete file

cortesep committed:
First reco workflow

mconcas committed:
[ITS-GPU] Fix warning in CI

@mpuccio FYI

sawenzel committed:
o2-sim: more shutdown improvements

* no need to kill multiple times
* no need to check sim-kinematics when one component errored

cortesep committed:
Bugs in reco tree

cortesep committed:
Corrected bug in Energy decoding

alibuild committed:
Please consider the following formatting changes

lucamicheletti93 committed:
Inclusion of CosTheta variable in the helicity reference frame  (#6261)

* Adding CosTheta Helicity

* Small fix for cosTheta Helicity

* Fixing issues in the CosTheta Helicity definition

* Fixing some issues with white spaces and clang

* Fixing some issues with white spaces and clang

* Fixing some issues with white spaces and clang (2)

* Fixing braces issue

* Fixing braces issue (2)

Co-authored-by: Luca Micheletti <lmichele@alicecerno2.cern.ch>

shahor02 committed:
Leave matched tracks at X last update point

In order to not bias the e.loss correction if it is secondary.

shahor02 committed:
Relate prim.tracks candidates to meanVertex applying mat.corr.

shahor02 committed:
fix in cascades casuality cuts, store cosPA

mario-krueger committed:
DPL Analysis: avoid instantiating HistogramRegistry functions per histogram name (#6323)

- allow to construct HistName objects from compile-time hashed names
- make HistName a nested class in HistogramRegistry since it has no meaning in global scope
- remove compile-time calculation of index in ConstStr<> since it is very registry specific

Co-authored-by: Mario Krüger <mario.kruger@cern.ch>

mario-krueger committed:
StepTHn: dont instantiate heavy heavy part of Fill() for each template (#6327)

Co-authored-by: Mario Krüger <mario.kruger@cern.ch>

afurs committed:
FITRaw: common digit2raw procedure is applied(implemented only for FV0) (#6322)

* Some cosmetics for FITRaw lib, CRTP removed from DigitBlock classes

* FIT: new features in FITRaw lib, common digit2raw converter added

* FIT: raw lib bugfix

* FITRaw: some cosmetics

* FIT: some cosmetics for FITRaw and bugfix(pythia8pp) for test exectutables

* hotfix for CI/CD

* Update DigitBlockBase.h

* Update CMakeLists.txt

* Update CMakeLists.txt

* Update DigitBlockBase.h

shahor02 committed:
Apply misalignment after building geometry in o2sim (#6329)

Co-authored-by: shahoian <ruben.shahoyan@cern.ch>

sawenzel committed:
o2-sim: Stability fix

Trying to address failures in the CI / lxplus-CVFMS, by:

* cleanup up sockets later
* giving more time to initialize (probably reading from CVMFS
  can slow down the init-process)
* set alignment to none by default as Geant4 crashes on the Mac when this
  is switched on

ktf committed:
DPL Analysis: extern declaration for common table types (#6288)

Should speed up compilation a bit.

cortesep committed:
TDC calibration object

davidrohr committed:
FST: Make GPU memory size settable in FST

davidrohr committed:
GPU: Enforce the CUDA Architecture we set, otherwise some other CMake module might override it

njacazio committed:
PID: Check that the TOF exp mom is positive (#6280)

shahor02 commented:
@cortesep There are conflicts to resolve before the CI tests can be started

cortesep committed:
ccdbHost for reconstruction + bux fixes

alibuild committed:
Please consider the following formatting changes

cortesep committed:
Correcting whitespaces

cortesep committed:
Correcting whitespaces

cortesep committed:
Correcting whitespaces

cortesep committed:
Enable debug output/printout from command line

alibuild committed:
Please consider the following formatting changes

shahor02 commented on Detectors/ZDC/reconstruction/src/DigiReco.cxx:
```diff
@@ -328,7 +331,7 @@ int DigiReco::reconstruct(int ibeg, int iend)
     for (int itdc = 0; itdc < NTDCChannels; itdc++) {
-      if (rec.fired[itdc] != 0x0) {
+      if (mVerbosity >= DbgFull && rec.fired[itdc] != 0x0) {
         printf("%d %u.%u TDC %d %x", ibun, rec.ir.orbit, rec.ir.bc, itdc, rec.fired[itdc]);
```
can you use LOG instead of printf? 
Also, putting `if  (mVerbosity..)` in a tight loops over the channels will cost some performance. Since this is a debug mode for local tests only, why don't you put these outputs in e.g.
#ifdef _ZDC_DEBUG_
#endif
guards, with the debug disabled by default?

cortesep replied:
I didn't know I was allowed to use ifdefs. I will follow your suggestion

cortesep replied:
I prefer to use printf for certain debug output. I protected them with #ifdef O2_ZDC_DEBUG

shahor02 replied:
once you use it only in your local tests, this is up to you. Just note that you can have with `LOGF(INFO, "printf formats",args...)` and `LOGP(INFO, "fmt formats", args...)` the printf and fmt formated outputs resp.

cortesep committed:
Corrections

cortesep committed:
Corrections

alibuild committed:
Please consider the following formatting changes

cortesep committed:
Configuration parameters for reconstruction

cortesep committed:
Configuration parameters for reconstruction - bug fixes

alibuild committed:
Please consider the following formatting changes

cortesep committed:
Configuration parameters for reconstruction - bug fixes

shahor02 committed:
fixes

shahor02 commented:
@cortesep 

There were lot of problems in your layout:
What is the ZDCEnergyParam? It has a header, mentioned in the LinkDef, but has not cxx and not compiled. Commenting it at the moment.
You had lot of other stuff with ClassDefNV, mentioned in the LinkDef but not in the CMakeFiles o2_target_root_dictionary

I've opened a PR in your repo, with it the object is read correctly.

PS Use `#include "Framework/Logger.h"` instead of `#include "FairLogger.h"`

cortesep commented:
Thanks for the corrections Ruben

cortesep committed:
Energy and Tower calibration

cortesep committed:
Correcting spaces

alibuild committed:
Please consider the following formatting changes

cortesep committed:
Correcting tabs

cortesep committed:
Inclusion of Energy and Tower calibration objects

cortesep committed:
Energy calibration and tower intercalibration in reconstruction

cortesep committed:
Correcting tabs

cortesep committed:
Merging

cortesep committed:
Whitespaces

shahor02 commented:
@cortesep it seems you have merged instead of rebasing to dev. 
Please rever your merge commit: `git revert 297b7fc08bd55dd4d5d141691834aa7f59202d29`
then do `git rebase dev`

cortesep commented:
I cannot revert. I get this error

git revert 297b7fc08bd55dd4d5d141691834aa7f59202d29
error: Commit 297b7fc08bd55dd4d5d141691834aa7f59202d29 is a merge but no -m option was given.
fatal: revert failed

cortesep committed:
Revert "Correcting tabs"

This reverts commit d7bad60056f521e05e6af555b4eeeaad282c79e5.

cortesep committed:
Correcting tabs and printout for lonely bunches

shahor02 commented:
@cortesep :  then do 
````
git reset --hard d7bad60056f521e05e6af555b4eeeaad282c79e5
````
i.e. reset to your last commit before the merge. And try to avoid merge in future...

cortesep committed:
Correcting tabs and printout for lonely bunches

cortesep committed:
Correcting tabs and printout for lonely bunches

cortesep commented:
Tested reconstruction with:
- pythia8 simulation
- digit2raw
- raw2digit
- reconstruction
Output looks reasonable

shahor02 commented:
OK, thanks, will merge once you will cleanup the wrong merge, rebase to dev, forcepush and it passes the checks.

cortesep commented:
It is very difficult to recover from the merge done by mistake. I'm closing this pull request and opening a new one from a fresh tree.
